# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/generic/DocAccessible.h
# Commit: 7b00a16ad21e
# Full Hash: 7b00a16ad21ee2b820b679199ae3806250d861e9
# Author: Nathan LaPre <nlapre@mozilla.com>
# Date: 2024-09-10 09:03:27
# Regressor Bug: 1794974
# File Overlap Count: 1
# Description:
#   Bug 1794974: Part 2: Add ability to query and send individual cache domains, r=Jamie
#   
#   This revision adds the capability of querying and sending information about
#   individual cache domains. It introduces the concept of active cache domains to
#   the accessibility service: cache domains that we think clients need. Conversely,
# ==============================================================================

diff -r 642b6d8c4d3b -r 7b00a16ad21e accessible/generic/DocAccessible.h
--- a/accessible/generic/DocAccessible.h	Mon Sep 09 23:02:19 2024 +0000
+++ b/accessible/generic/DocAccessible.h	Mon Sep 09 23:02:19 2024 +0000
@@ -8,6 +8,7 @@
 
 #include "HyperTextAccessible.h"
 #include "AccEvent.h"
+#include "nsAccessibilityService.h"
 
 #include "nsClassHashtable.h"
 #include "nsTHashMap.h"
@@ -118,7 +119,8 @@
    * Note that this CANNOT be used for anything which fires events, since events
    * must be fired after their associated cache update.
    */
-  void QueueCacheUpdate(LocalAccessible* aAcc, uint64_t aNewDomain);
+  void QueueCacheUpdate(LocalAccessible* aAcc, uint64_t aNewDomain,
+                        bool aBypassActiveDomains = false);
 
   /**
    * Walks the dependent ids and elements maps for the given accessible and
@@ -559,8 +561,10 @@
    * Called from NotificationController to process this doc's
    * queued cache updates. For each acc in the map, this function
    * sends a cache update with its corresponding CacheDomain.
+   * Each domain bit in aInitialDomains indicates that this is the first push
+   * for that cache domain.
    */
-  void ProcessQueuedCacheUpdates();
+  void ProcessQueuedCacheUpdates(uint64_t aInitialDomains = 0);
 
   /**
    * Called from NotificationController before mutation events are processed to
@@ -817,6 +821,12 @@
   friend class EventTree;
   friend class NotificationController;
 
+  /*
+   * The accessibility service may need to process queued cache updates outside
+   * of the regular NotificationController flow.
+   */
+  friend class ::nsAccessibilityService;
+
  private:
   void SetRoleMapEntryForDoc(dom::Element* aElement);
 